llvm/lib/Transforms/Scalar/NewGVN.cpp

```diff
@@ -653,6 +658,16 @@ class NewGVN {
   // Deletion info.
   SmallPtrSet<Instruction *, 8> InstructionsToErase;
 
+  // Map candidate load to their depending instructions.
+  mutable std::map<Value *, DenseSet<std::pair<Instruction *, BasicBlock *>>>
```

Comment:
It is needed because the map is written in const function which is called by another const function. It seems that it is the cleaner way to deal with this situation.